### PR TITLE
Improve compiler/execution/program output

### PR DESCRIPTION
**Pull request by badeip**
Make information more compact to allow for more program output. 
Do not print compiler return code on successful execution
Print console output with monospaced font
Print console stderr with red colour

### CODE REVIEW

File: static/output.js [hunk 1]
```diff
@@ -59,15 +59,19 @@ define(function (require) {
...
-
-        if (result.execResult) {
-            var elem = $('<div><hr></div>').appendTo(this.contentRoot);
-            this.add("Output from execution:");
-            _.each((result.execResult.stdout || []).concat(result.execResult.stderr || []), function (obj) {
-                this.add(obj.text, obj.line);
-            }, this);
-            this.add("User code exited with result code " + result.execResult.code);
+        if (!result.execResult) {
+            this.add("compiler returned: " + result.code);
```
Comment (RabsRincon) [inline static/output.js]: This should probably be properly capitalized

File: static/output.js [hunk 2]
```diff
@@ -59,15 +59,19 @@ define(function (require) {
...
+        } else {
+            this.add("program returned: " + result.execResult.code);
```
Comment (RabsRincon) [inline static/output.js]: This should probably be properly capitalized

File: static/output.js [hunk 3]
```diff
@@ -77,6 +81,14 @@ define(function (require) {
         if (id === this.compilerId) this.fontScale.setScale(scale);
     };
 
+    Output.prototype.programOutput = function (msg, color) {
+        var elem = $('<div></div>').appendTo(this.contentRoot);
```
Comment (RabsRincon) [inline static/output.js]: I think you can chain the `.text` and `.css` methods with this line. Keep them in separate lines, but there's no need to `elem.` before each one

File: static/output.js [hunk 4]
```diff
@@ -59,15 +59,19 @@ define(function (require) {
             this.add(obj.text, obj.tag ? obj.tag.line : obj.line);
         }, this);
 
-        this.add("Compiler exited with result code " + result.code);
```
Comment (RabsRincon) [inline static/output.js]: I'm not too sure about the removal of the result code. What was your motivation behind it? I do think it is useful even in the "success" case

**Comment by RabsRincon** [comment]
Hello! Thank you for this, I'm happy that you've taken the time to improve the (not the best) UI!

File: static/output.js (hunk 4 above)
Comment (badeip) [reply static/output.js]: My reasoning was that the program will not execute unless the compilation return code is 0, so that is redundant information.

**Comment by badeip** [comment]
Javascript is not really my domain, so do what you want with this :)
This was just some changes I felt that somewhat improves execution output readability.

File: static/output.js (hunk 4 above)
Comment (RabsRincon) [reply static/output.js]: But what will happen if no execution is requested? You now dont have what the compiler returned, right?

**Comment by mattgodbolt** [comment]
Thanks so much! I'm happy to drop the compiler exiting with 0 part, as you say. @RabsRincon if you're cool with this let's merge! :)

**Pull request merged**